### PR TITLE
box: check tuple_field_by_path first argument type

### PR DESCRIPTION
**Pull request by ImeevMA**
This patch adds a type check of the first argument of the tuple_field_by_path() function.

Closes tarantool/security#82

NO_DOC=internal
NO_CHANGELOG=internal

### CODE REVIEW

File: src/box/lua/tuple.c [hunk 1]
```diff
@@ -649,7 +649,7 @@ lbox_tuple_transform(struct lua_State *L)
 static int
 lbox_tuple_field_by_path(struct lua_State *L)
 {
-	struct tuple *tuple = luaT_istuple(L, 1);
+	struct tuple *tuple = luaT_checktuple(L, 1);
```
Comment (locker) [inline src/box/lua/tuple.c:652]: Nit: The assertion for `tuple` is no longer needed. Let's remove.

Let's also remove the assertion for `path` and instead write:

```C
const char *path = luaL_checklstring(L, 2, &len);
```

File: test/box-luatest/func_test.lua [hunk 2]
```diff
@@ -120,3 +120,12 @@ g.test_gh_7822_vfunc_format = function()
         t.assert_equals(box.space._vfunc:format(), box.space._func:format())
     end)
 end
+
+g.test_ghs_82_tuple_field_by_path = function()
```
Comment (locker) [inline test/box-luatest/func_test.lua:124]: Why do you think we should add this test case to this file? They look completely unrelated.

BTW I think that it's okay to commit this fix without a test, because testing internal functions isn't really necessary.

**Review by locker — CHANGES_REQUESTED** [review]

File: src/box/lua/tuple.c (hunk 1 above)
Comment (ImeevMA) [reply src/box/lua/tuple.c:652]: Fixed.

File: test/box-luatest/func_test.lua (hunk 2 above)
Comment (ImeevMA) [reply test/box-luatest/func_test.lua]: I decided it because of the test-file name.

Dropped test.

**Review by locker — APPROVED** [review]

**Comment by locker** [comment]
Cherry-picked to 2.10.
